test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_svm_mem.cpp

```diff
@@ -0,0 +1,413 @@
+//
+// Copyright (c) 2024 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "basic_command_buffer.h"
+#include "svm_command_basic.h"
+#include "procs.h"
+
+
+//--------------------------------------------------------------------------
+namespace {
+
+// CL_INVALID_COMMAND_QUEUE if command_queue is not NULL.
+struct CommandBufferCommandSVMQueueNotNull : public BasicSVMCommandBufferTest
+{
+    using BasicSVMCommandBufferTest::BasicSVMCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandSVMMemcpyKHR(
+            command_buffer, queue, svm_out_mem(), svm_in_mem(), data_size(), 0,
+            nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_COMMAND_QUEUE,
+            "clCommandSVMMemcpyKHR should return CL_INVALID_COMMAND_QUEUE",
+            TEST_FAIL);
+
+        error = clCommandSVMMemFillKHR(command_buffer, queue, svm_in_mem(),
+                                       &pattern_1, sizeof(cl_char), data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_COMMAND_QUEUE,
+            "clCommandSVMMemFillKHR should return CL_INVALID_COMMAND_QUEUE",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        return BasicSVMCommandBufferTest::Skip()
+            || is_extension_available(device,
+                                      "cl_khr_command_buffer_multi_device");
```

Comment:
Not sure we need this check, we'll have only set a single command-queue when `command_buffer` was created. So I don't think we satisfy the error wording for `CL_INVALID_COMMAND_QUEUE` and would remove this multi-device extension check.


[CL_INVALID_COMMAND_QUEUE](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_INVALID_COMMAND_QUEUE) if the [cl_khr_command_buffer_multi_device](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#cl_khr_command_buffer_multi_device) extension is supported; and either command_queue is NULL and command_buffer was created with more than one queue, or command_queue is not NULL and not a command-queue listed on command_buffer creation.

Reply:
@EwanC so I think this Skip() could be removed, but there could be case it extension is supported, command_queue it not null but command_buffer is created with different command_queue. What do you think about it?

Reply:
I think it could be a valid test, but I think its scope fits better for when we get round to writing cl_khr_command_buffer_multi_device tests, I've made a note in https://github.com/KhronosGroup/OpenCL-CTS/issues/1703